File: backend/semgrep_rules/tests/python/kisa_sw_11_csrf.py
from django.views.decorators.csrf import csrf_exempt

# ruleid: kisa.sw11.python.csrf-protection-disabled
@csrf_exempt
def unsafe(request):
    return "changed"

def safe(request):
    # ok: kisa.sw11.python.csrf-protection-disabled
    return "protected"

File: backend/semgrep_rules/tests/java/kisa_sw_11_csrf.java
class kisa_sw_11_csrf {
    void unsafe(Object http) {
        // ruleid: kisa.sw11.java.csrf-protection-disabled
        http.csrf().disable();
    }
    void safe(Object http) {
        // ok: kisa.sw11.java.csrf-protection-disabled
        http.csrf();
    }
}
